File: core/middleware/cors.py
from fastapi.middleware.cors import CORSMiddleware
from fastapi import FastAPI

from config.settings import settings


def setup_cors(app: FastAPI):
    app.add_middleware(
        CORSMiddleware,
        allow_origins=settings.cors.allow_origins,
        allow_methods=settings.cors.allow_methods,
        allow_headers=settings.cors.allow_headers,
        allow_credentials=settings.cors.allow_credentials,
    )

File: core/middleware/cors.go
package middleware

import (
	"hei-gin/config"

	"github.com/gin-contrib/cors"
	"github.com/gin-gonic/gin"
)

// CORS returns a Gin middleware that configures Cross-Origin Resource Sharing.
func CORS() gin.HandlerFunc {
	return cors.New(cors.Config{
		AllowOrigins:     config.C.CORS.AllowOrigins,
		AllowMethods:     config.C.CORS.AllowMethods,
		AllowHeaders:     config.C.CORS.AllowHeaders,
		AllowCredentials: config.C.CORS.AllowCredentials,
	})
}
